### PR TITLE
Dynamo irsa

### PR DESCRIPTION
**Pull request by candonov**
### What does this PR do?

Adds dynamodb+irsa example

### Motivation

dynamodb tables creating is fast (a few seconds) which makes them great for demos

### More

- [X] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)

- [X] Yes, I have added a new example under [examples](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/examples) to support my PR

- [X] Yes, I have updated the [docs](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/doc) for this feature

### CODE REVIEW

File: compositions/upbound-aws-provider/dynamo-irsa/definition.yaml [hunk 1]
```diff
@@ -0,0 +1,163 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+apiVersion: apiextensions.crossplane.io/v1
+kind: CompositeResourceDefinition
+metadata:
```
Comment (nabuskey) [inline compositions/upbound-aws-provider/dynamo-irsa/definition.yaml:6]: Let's add labels.

File: compositions/upbound-aws-provider/sns/sns.yaml [hunk 2]
```diff
@@ -31,7 +31,7 @@ spec:
           toFieldPath: spec.forProvider.tags
           policy:
             mergeOptions:
-              appendSlice: true
+              keepMapValues: true
```
Comment (nabuskey) [inline compositions/upbound-aws-provider/sns/sns.yaml:34]: Thanks for making these changes. Definitely needed. In the future though, let's do it in a separate PR

File: examples/upbound-aws-provider/composite-resources/databases/dynamo-irsa/claim/dynamo-irsa.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+apiVersion: awsblueprints.io/v1alpha1
+kind: DynamoIRSA
+metadata:
+  name: dynamo-irsa-test
+spec:
+  writeConnectionSecretToRef:
```
Comment (nabuskey) [inline examples/upbound-aws-provider/composite-resources/databases/dynamo-irsa/claim/dynamo-irsa.yaml:6]: Use `publishConnectionDetailsTo` instead.

**Review by nabuskey — CHANGES_REQUESTED** [review]

File: examples/upbound-aws-provider/composite-resources/databases/dynamo-irsa/claim/dynamo-irsa.yaml (hunk 3 above)
Comment (candonov) [reply examples/upbound-aws-provider/composite-resources/databases/dynamo-irsa/claim/dynamo-irsa.yaml:6]: I tried had issues with it. Atm all the compositions in the upbound-aws-provider section use `writeConnectionSecretToRef`. I will change at a later time across the board.

File: compositions/upbound-aws-provider/dynamo-irsa/definition.yaml (hunk 1 above)
Comment (candonov) [reply compositions/upbound-aws-provider/dynamo-irsa/definition.yaml:6]: What labels and why? The rest of the XRDs don't have labels.

File: compositions/upbound-aws-provider/sns/sns.yaml (hunk 2 above)
Comment (candonov) [reply compositions/upbound-aws-provider/sns/sns.yaml:34]: will do!

File: compositions/upbound-aws-provider/dynamo-irsa/definition.yaml (hunk 1 above)
Comment (nabuskey) [reply compositions/upbound-aws-provider/dynamo-irsa/definition.yaml:6]: my bad I was looking at the wrong file

File: examples/upbound-aws-provider/composite-resources/databases/dynamo-irsa/claim/dynamo-irsa.yaml (hunk 3 above)
Comment (nabuskey) [reply examples/upbound-aws-provider/composite-resources/databases/dynamo-irsa/claim/dynamo-irsa.yaml:6]: What issue did you have? It does work and we really should be using it for new compositions.

**Review by nabuskey — APPROVED** [review]
LGTM

**Pull request merged**